services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/OVSGoalStateServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.dataplane.service.impl;
+
+import com.futurewei.alcor.common.message.MessageClient;
+import com.futurewei.alcor.dataplane.config.Config;
+import com.futurewei.alcor.dataplane.config.grpc.GoalStateProvisionerClient;
+import com.futurewei.alcor.dataplane.service.GoalStateService;
+import com.futurewei.alcor.schema.Goalstate;
+import com.futurewei.alcor.schema.Goalstateprovisioner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+@Service
+public class OVSGoalStateServiceImpl implements GoalStateService {
+  private static final Logger LOG = LoggerFactory.getLogger(OVSGoalStateServiceImpl.class);
+
+  private int port;
+  private String ip;
+
+  public MessageClient getKafkaClient() {
+    return kafkaClient;
+  }
+
+  MessageClient kafkaClient = null;
+  ExecutorService executorService = Executors.newCachedThreadPool();
+
+  @Override
+  public List<List<Goalstateprovisioner.GoalStateOperationReply.GoalStateOperationStatus>>
+      SendGoalStateToHosts(
+          Map<String, Goalstate.GoalState> gss, boolean isFast, int port, boolean isOvs) {
+
+    if (isOvs) {
+      List<List<Goalstateprovisioner.GoalStateOperationReply.GoalStateOperationStatus>> result =
+          new ArrayList<>();
+
+      gss.entrySet()
+          .parallelStream()
```

Comment:
Like this!

Reply:
We do need to think of using async grpc channels especially when we try to push to multiple hosts at the same time. An old issue https://github.com/futurewei-cloud/alcor/issues/58 has been created for tracking purpose.

Reply:
for async mode, aca needs to support , we could leave it like this since ACA not supporting async mode

Reply:
That is right. I will assign #58 to you and @er1cthe0ne. We can discuss the design and priority together.

Reply:
ok